lib/ansible/modules/storage/purestorage/purefa_user.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2018, Simon Dodsley (simon@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: purefa_user
+version_added: '2.8'
+short_description: Create, modify or delete FlashArray local user account
+description:
+- Create, modify or delete local users on a Pure Stoage FlashArray.
+author:
+- Simon Dodsley (@sdodsley)
+options:
+  state:
+    description:
+    - Create, delete or update `local user account
+    default: present
+    choices: [ absent, present, update ]
```

Comment:
What's the difference between

Seeing state `present` and state `update` looks a little strange here.

```yaml
- name: Change role type for existing user
  purefa_user:
    name: ansible
    role: storage_admin
    state: present # <----
    fb_url: 10.10.10.2
    api_token: e31060a7-21fc-e277-6240-25983c6c4592
```


```yaml
- name: Change role type for existing user
  purefa_user:
    name: ansible
    role: storage_admin
    state: update # <----
    fb_url: 10.10.10.2
    api_token: e31060a7-21fc-e277-6240-25983c6c4592
```

Reply:
`present` is for the creation of a user, where you specify a specific role type. `update` can be used to change this, however, need to split this out into a separate section as `update` can also be used to change a password or force an API token change. Your example won't do anything as you create a user with `role: storage_admin` and then try and change to the same value. There is a check for this eventuality. 
The examples show creating auser with `readonly` and then a change to a different role.

Reply:
Just fixed line 26 with a weird character in the description

Reply:
Generally for Ansible modules `state: present` will update the object to match what's in the task.

Reply:
@gundalow I have refactored this to only use `present` and `absent`.